### PR TITLE
Make code style consistent with clang-format

### PR DESCRIPTION
**Pull request by steple**
Add a .clang-format file with settings that are picked based on Dave's desire for Allman braces and tabs of width 8 (set your editor to tab width of 8 for the best experience).  Some of the other settings have been picked  to preserve the current style when possible.

Also mark some parts of the code as off-limits for clang-format to preserve readability of table-like lists of registers.

To run clang-format you can use
clang-format-9 -i --style=file *.{h,c} tools/double.c

No functional changes.

### CODE REVIEW

File: raspiraw.c [hunk 1]
```diff
@@ -290,104 +310,115 @@ typedef struct raspiraw_params {
...
 
 	return 0;
 }
 
-const struct sensor_def * probe_sensor(int fd)
+const struct sensor_def *probe_sensor(int fd)
 {
 	const struct sensor_def **sensor_list = &sensors[0];
-	const struct sensor_def *sensor = NULL;
+	const struct sensor_def *sensor	      = NULL;
```
Comment (6by9) [inline raspiraw.c:403]: Indenting before the =.

File: raspiraw.c [hunk 2]
```diff
@@ -216,57 +218,75 @@ enum {
...
-	{ CommandDecodeMetadata,"-metadata",	"m","Decode register metadata", 0 },
-	{ CommandAwb,		"-awb",		"awb","Use a simple grey-world AWB algorithm", 0 },
-	{ CommandNoPreview,	"-nopreview",	"n",  "Do not send the stream to the display", 0 },
-	{ CommandProcessing,	"-processing",	"P",  "Pass images into an image processing function", 0 },
-	{ CommandPreview,	"-preview",	"p",  "Preview window settings <'x,y,w,h'>", 1 },
-	{ CommandFullScreen,	"-fullscreen",	"fs", "Fullscreen preview mode", 0 },
-	{ CommandOpacity,	"-opacity",	"op", "Preview window opacity (0-255)", 1},
-	{ CommandProcessingYUV,	"-processing_yuv", "PY",  "Pass processed YUV images into an image processing function", 0 },
-	{ CommandOutputYUV,	"-output_yuv",  "oY", "Set the output filename for YUV data", 0 },
+static COMMAND_LIST cmdline_commands[] = {
```
Comment (6by9) [inline raspiraw.c:221]: Losing the tabulation of entries here makes it less readable.

And the space after {, and before } also help make it readable.

File: raspiraw.c [hunk 3]
```diff
@@ -290,104 +310,115 @@ typedef struct raspiraw_params {
...
+		.buf   = values,
+	    },
 	};
 
 	if (sensor->i2c_addressing == 1)
 	{
 		msgs[0].len = 1;
 	}
-	msgset.msgs = msgs;
+	msgset.msgs  = msgs;
```
Comment (6by9) [inline raspiraw.c:388]: Why?

File: raspiraw.c [hunk 4]
```diff
@@ -560,45 +607,54 @@ static void buffers_to_rawcam(RASPIRAW_CALLBACK_T *dev)
...
+		if (!(buffer->flags & MMAL_BUFFER_HEADER_FLAG_CODECSIDEINFO) &&
+		    (((count++) % cfg->saverate) == 0))
 		{
 			// Save every Nth frame
 			// SD card access is too slow to do much more.
 			FILE *file;
 			char *filename = NULL;
-			if (create_filenames(&filename, cfg->output, count) == MMAL_SUCCESS)
+			if (create_filenames(&filename, cfg->output, count) ==
+			    MMAL_SUCCESS)
```
Comment (6by9) [inline raspiraw.c:634]: Unnecessary line break. Do we have a max line length param that can be tweaked upwards? Even the kernel coding standard now allows up to 120 chars.

File: raspiraw.c [hunk 5]
```diff
@@ -607,51 +663,61 @@ static void callback(MMAL_PORT_T *port, MMAL_BUFFER_HEADER_T *buffer)
...
+			MMAL_BUFFER_HEADER_T *out =
+			    mmal_queue_get(dev->isp_ip_pool->queue);
 			if (out)
 			{
-				//vcos_log_error("replicate buffer %p for isp", buffer);
+				// vcos_log_error("replicate buffer %p for isp",
+				// buffer);
 				mmal_buffer_header_replicate(out, buffer);
-				out->data = buffer->data;
+				out->data	= buffer->data;
```
Comment (6by9) [inline raspiraw.c:692]: More weird indentation before an assignment.

**Review by 6by9 — COMMENTED** [review]
This is why I don't like trying to totally dictate a coding style, but also realise others find it helpful. It's quite a disruptive change so I have some hesitation.

I think there must be a few further setting to control:
- this funny indentation with assignments
- retain the tabulation of the config options
- allow for a longer line length (120 or 150 cols)

and that will deal with a lot of my issues. The kernel coding style does most of the above, so the .clang_format in https://github.com/torvalds/linux/blob/master/.clang-format probably include the relevant runes.

**Review by steple — COMMENTED** [review]
Changed line length, alignment of consecutive assignments, and add spaces to {} lists.

File: raspiraw.c (hunk 2 above)
Comment (steple) [reply raspiraw.c:218]: Fixed.

**Comment by steple** [comment]
And if you'd like to see what would happen if we just used the kernel .clang-format file straight up, check https://github.com/steple/raspiraw/compare/master...steple:kernel_style

**Comment by steple** [comment]
@6by9 what do you think?

After this I can post my change to enable crop on the imx219, which is what I'm really after.

**Comment by steple** [comment]
Ping.

**Comment by 6by9** [comment]
Sorry for the delay.
I haven't checked it line by line, but it looks reasonable.

**Comment by 6by9** [comment]
And it's not fair that I'm about to make a load of changes to replace MMAL with V4L2 without merging this first :-)

Thanks.

**Comment by steple** [comment]
Thanks, this allowed me to open #54 which is what I was really after.

**Pull request merged**